### DISCUSSION TITLE
Popup size isn't constrained to containing page

### DISCUSSION THREAD

**Discussion by BioTurboNick** [CommunityToolkit/Maui#761] [Q&A] [answered]
```
<Grid />
```

Big central popup well-proportioned.
![image](https://user-images.githubusercontent.com/1438610/201560482-31ae2fe3-11c8-4781-9cbd-200551b5ef20.png)

```
<Grid>
        <Label Text="{Binding Title}" />
</Grid>
```

One label causes the popup to shrink to fit
![image](https://user-images.githubusercontent.com/1438610/201560049-3e43574c-6929-4a23-ae11-82bcb6260109.png)

```
<Grid RowDefinitions="*,*,*">
        <Label Text="{Binding Title}" />

        <Label Grid.Row="1"
               Text="{Binding Message}" />
    </Grid>
```

Adding a long label causes it to extend well beyond the screen.
![image](https://user-images.githubusercontent.com/1438610/201560207-00a89e7e-0c9b-482f-b172-0881d5898a5c.png)

On Android, if it matters.

The documentation suggests sizing should automatically fit unless HorizontalOptions or VerticalOptions are customized, which I am not. I'm using `Shell.Current.CurrentPage.ShowPopupAsync()` if that matters. https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/views/popup

Am I missing something, or is this a bug?

**Comment by mhrastegari** [comment]
I believe there is a `Size` property like this `<mct:Popup Size="300,300" />`.

**Reply by BioTurboNick** [reply]
> I believe there is a `Size` property like this `<mct:Popup Size="300,300" />`.

Yes, the point is the default isn't behaving as documented.

**Comment by pictos** [comment, marked as answer]
@BioTurboNick, the calculation is based on how each layout calculates itself and takes that as the actual size for the Popup. I grabbed this phrase from the docs:

> The Popup will always try to constrain the actual size of the Popup to the size of the View unless a Size is specified.

As you can see it `tries` to constrain.

**Reply by BioTurboNick** [reply]
I suppose that could technically be true, but not being able to handle long text in a label seems like it isn't trying very hard. A Page constrains its children to the screen just fine, why does Popup have this problem?

**Reply by pictos** [reply]
Well, the bad part is we're a project maintained by community members in our spare time. So we don't have too much resources to test the implementation in all possible scenarios. The good is this project is open source, so you can help us improve this lib sending a PR 😉 let me know if you need any guidance in forking the repo and submit your first PR.

**Comment by CliffAgius** [comment]
Closed as answered
